extensions/iam/oauth2/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/impl/IdentityProviderKeyResolver.java

```diff
@@ -105,10 +106,10 @@ public void refreshKeys() {
      */
     Map<String, RSAPublicKey> deserializeKeys(List<JwkKey> jwkKeys) {
         Map<String, RSAPublicKey> newKeys = new HashMap<>();
-
+ 
         // instantiate the RSA public keys by kid
         for (JwkKey key : jwkKeys) {
-            if (!"RSA".equals(key.getKty()) || !"sig".equals(key.getUse())) {
+            if (!"RSA".equals(key.getKty())) {
                 continue;
```

Comment:
I would avoid the usage of the `continue` statement, better to refactor this for loop using the `Stream` approach, that `"RSA".equals(key.getKty())` condition would be simply defined as a filter.

Reply:
done